tools/dep-compare.py

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env python3
+
+import toml
+from collections import defaultdict
+
+# Given a Cargo.lock path, get a dictionary mapping package names to all version numbers for that package
+def load_cargo_lock_packages(path):
+    lock = toml.load(path)
+    result = defaultdict(set)
+    for package in lock['package']:
+        result[package['name']].add(package['version'])
+    return result
+
+root = load_cargo_lock_packages("Cargo.lock")
+consensus = load_cargo_lock_packages("consensus/enclave/trusted/Cargo.lock")
+fog_ingest = load_cargo_lock_packages("fog/ingest/enclave/trusted/Cargo.lock")
+fog_ledger = load_cargo_lock_packages("fog/ledger/enclave/trusted/Cargo.lock")
+fog_view = load_cargo_lock_packages("fog/view/enclave/trusted/Cargo.lock")
```

Comment:
Sounds like we want this script to do several things, and this should go into a function which automatically searches out enclave folders and does conditional checks against the deps (which the script runner could specify at the time)